docs/scripts/util/checkLink.ts

```diff
@@ -0,0 +1,87 @@
+import https from 'https';
+import http from 'http';
+import { IGNORED_LINKS } from '../../src/data/ignoredLinks';
+
+const requestedUrl: Set<string> = new Set();
+export async function checkLink(
+  {
+    href,
+    tagName,
+    tagText,
+    pageIdx,
+    pageUrl,
+  }: {
+    href: string;
+    tagName: string;
+    tagText: string;
+    pageIdx: string;
+    pageUrl: string;
+  },
+  linkIdx: number
+) {
+  return new Promise(async (res, rej) => {
+    if (!href) {
+      console.log(
+        `⚠️[WARNING...] page #${pageIdx} link #${linkIdx} "${tagName}" tag "${tagText}" doesn't have a href.`
+      );
+      res(0);
+    } else if (IGNORED_LINKS.includes(href) || requestedUrl.has(href)) {
+      console.log(
+        `⏭[SKIPPING...] page #${pageIdx} link #${linkIdx} ${href} from ${tagName} tag "${tagText}" on page ${pageUrl}, because it is on the IGNORED_LINKS list or have already requested.`
+      );
+      res(0);
+    } else if (href.includes('https:')) {
+      const request = await https.get(href, async ({ statusCode = 0 }) => {
+        await returnStatus({ statusCode, href });
+        requestedUrl.add(href);
+        res(statusCode);
+      });
+      request.end();
+    } else {
+      const request = await http.get(href, async ({ statusCode = 0 }) => {
+        await returnStatus({ statusCode, href });
+        requestedUrl.add(href);
+        res(statusCode);
+      });
+      request.end();
```

Comment:
`http.get` vs `https.get ` seems to be the only difference between these blocks can we consolidate?
```
} else {
const { request: httpRequest } = href.includes('https:') ? https : http;
const request = await httpRequest.get(href, async ({ statusCode = 0 }) => {
        await returnStatus({ statusCode, href });
        requestedUrl.add(href);
        res(statusCode);
      });
      request.end();
}
```

Reply:
Good point. I'll do that